org.mosh.mosh.json

```diff
@@ -0,0 +1,129 @@
+{
+  "app-id": "org.mosh.mosh",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "18.08",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "mosh",
+  "separate-locales": false,
+  "finish-args": [
+    "--share=network",
+    "--talk-name=org.freedesktop.Flatpak"
+  ],
+  "cleanup": [
+    "*.a",
+    "*.la",
+    "/include",
+    "/lib/pkgconfig",
+    "/share/man"
+  ],
+  "modules": [
+    {
+      "name": "protobuf",
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://github.com/google/protobuf/releases/download/v3.8.0/protobuf-all-3.8.0.tar.gz",
+          "sha256": "b7220b41481011305bf9100847cf294393973e869973a9661046601959b2960b"
+        }
+      ],
+      "cleanup": [
+        "/bin/protoc"
+      ]
+    },
+    {
+      "name": "perl",
+      "no-autogen": true,
+      "config-opts": [
+        "-des"
+      ],
+      "cleanup": [
+        "/bin/perl5*",
+        "/bin/c2ph",
+        "/bin/corelist",
+        "/bin/cpan",
+        "/bin/enc2xs",
+        "/bin/encguess",
+        "/bin/h2*",
+        "/bin/instmodsh",
+        "/bin/json_pp",
+        "/bin/libnetcfg",
+        "/bin/perlbug",
+        "/bin/perldoc",
+        "/bin/perlivp",
+        "/bin/perlthanks",
+        "/bin/piconv",
+        "/bin/pl2pm",
+        "/bin/pod*",
+        "/bin/prove",
+        "/bin/pstruct",
+        "/bin/ptar*",
+        "/bin/shasum",
+        "/bin/splain",
+        "/bin/xsubpp",
+        "/bin/zipdetails",
+        "/lib/perl5/site_perl",
+        "/man",
+        "*.h",
+        "*.pod"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://www.cpan.org/src/5.0/perl-5.28.0.tar.gz",
+          "sha256": "7e929f64d4cb0e9d1159d4a59fc89394e27fa1f7004d0836ca0d514685406ea8"
+        },
+        {
+          "type": "script",
+          "dest-filename": "configure",
+          "commands": [
+            "exec ./configure.gnu $@"
+          ]
+        }
+      ],
+      "post-install": [
+        "find /app/lib/perl5 -type f -exec chmod u+w {} \\;"
+      ]
+    },
+    {
+      "name": "mosh",
+      "config-opts": [
+        "--disable-server"
```

Comment:
Does the server introduce additional dependencies, or is it just™ a matter of:

* The client expects `mosh-server` to be in the path (workaround: `org.mosh.mosh --server='flatpak run --command=mosh-server org.mosh.mosh'`)
* The server would need to be patched to spawn a shell with `flatpak-spawn --host`

Reply:
Yeah, these - although libutempter has setuid helpers for updating utmp - but it's optional. I just don't think that it's that worthwhile... :)